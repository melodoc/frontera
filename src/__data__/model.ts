// TODO: Move where, where used

export type Link = {
  href: string;
  label: string;
};
