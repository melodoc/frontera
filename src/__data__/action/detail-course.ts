import { getConfigValue } from '@ijl/cli';

import { init, success, error } from '../slice/detail-course'
import axios from "axios";

export const getDetailCourse = (id) => async (dispatch) => {
    dispatch(init());

    const baseApiUrl = getConfigValue('frontera.api');

    const response =  await axios(`${baseApiUrl}/detail-course/${id}`, {
        method: 'GET',
        headers: {
            'Content-Type': 'application/json;charset=utf-8',
        }
    });

    if (response.data) {
        dispatch(success(response.data.data));
    } else {
        dispatch(error('Ошибка!'));
    }
};

export default getDetailCourse;
