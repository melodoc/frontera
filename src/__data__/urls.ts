export const URLs = {
  auth: {
    url: '/auth'
  },
  personalization: {
    url: '/personalization'
  },
  catalogCourses: {
    url: '/catalog-courses'
  },
  home: {
    url: '/home'
  },
  coursePage: {
    url: '/course-page'
  },
  root: {
    url: '/'
  }
};
