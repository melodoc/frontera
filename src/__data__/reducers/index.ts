import { combineReducers } from 'redux';

import { reducer as suggestions } from '../slice/suggestions';
import { reducer as trending } from '../slice/trending';
import { reducer as themes } from '../slice/themes';
import { reducer as detailCourse } from '../slice/detail-course';

import { reducer as login } from '../slice/login';
import { reducer as registration } from '../slice/registration';

export default combineReducers({
    suggestions,
    trending,
    themes,
    detailCourse,
    login,
    registration,
});
