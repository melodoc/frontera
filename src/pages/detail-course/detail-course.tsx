import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

import { Head } from '../../components/head/head';
import { Header } from '../../components/header/header';
import { PageHeader } from '../../components/page-header';
import { CourseInfo } from '../../components/course-info/course-info';
import { CourseCards } from '../../components/course-cards/course-cards';
import { getDetailCourse } from '../../__data__/action/detail-course';
import { Loader } from '../../components/loader/loader';
import { useAppSelector } from '../../__data__/store/hooks';
import style from './detail-course.module.scss';

export const DetailCourse = ({ courseId }) => {
  const dispatch = useDispatch();
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const { data, isLoading } = useAppSelector((state) => ({
    data: state.detailCourse.data,
    isLoading: state.detailCourse.loading
  }));
  useEffect(() => {
    dispatch(getDetailCourse(courseId));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  if (data == null || !data) {
    return <Loader />;
  }

  return (
    <>
      <Head title="Details" isLoading={isLoading} />
      <Header />
      <main className={style.Container}>
        <PageHeader label={data.title} />
        <CourseInfo info={data.info} description={data.description} />
        <section className={style.gallery}>
          <CourseCards cards={data.videoList} />
        </section>
      </main>
    </>
  );
};
