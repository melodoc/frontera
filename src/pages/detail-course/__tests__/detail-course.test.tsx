import React from 'react';
import { describe, it, expect, beforeEach } from '@jest/globals';
import { mount } from 'enzyme';
import axios from 'axios';
import { Provider } from "react-redux";
import MockAdapter from "axios-mock-adapter";

import { multipleRequest } from "../../../test-utils";
import { store } from "../../../__data__/store";
import { DetailCourse } from "../detail-course";

// eslint-disable-next-line @typescript-eslint/no-var-requires
const detailCourseResponse = require('../../../../stubs/api/mocks/detail-course/success');

describe('Тестируем DetailCourse', () => {
    let mockApi;
    beforeEach(() => {
        mockApi = new MockAdapter(axios);
    });
    it('Тестируем рендер ', async () => {
        const courseId = 1;
        const component = mount(
            <Provider store={store}>
                <DetailCourse courseId={courseId} />
            </Provider>
        );
        expect(component).toMatchSnapshot();
        const response = [
            ['GET', `/detail-course/${courseId}`, {}, 200, { ...detailCourseResponse }]
        ];
        await multipleRequest(mockApi, response);
        component.update();
        expect(component).toMatchSnapshot();
    });
});