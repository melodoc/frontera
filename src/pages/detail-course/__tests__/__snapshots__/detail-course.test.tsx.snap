// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем DetailCourse Тестируем рендер  1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <DetailCourse
    courseId={1}
  >
    <Loader>
      <div
        className="spinner"
      >
        <div
          className="bounce1"
        />
        <div
          className="bounce2"
        />
        <div
          className="bounce3"
        />
      </div>
    </Loader>
  </DetailCourse>
</Provider>
`;

exports[`Тестируем DetailCourse Тестируем рендер  2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <DetailCourse
    courseId={1}
  >
    <Header>
      <div
        className="headerWrapper"
      >
        <header
          className="header"
        >
          <nav
            className="headerNav"
          >
            <IconLogo>
              <a
                className="navLogo"
                href=""
              >
                <img
                  alt="Purple Frontera logo"
                  src="FILE_MOCK"
                />
              </a>
            </IconLogo>
            <HeaderNav>
              <Link
                as={[Function]}
                to="/home"
              >
                <Link
                  className="link"
                  to="/home"
                >
                  <a
                    className="link"
                    href="/home"
                  >
                    Главная
                  </a>
                </Link>
              </Link>
              <Link
                as={[Function]}
                to="/catalog-courses"
              >
                <Link
                  className="link"
                  to="/catalog-courses"
                >
                  <a
                    className="link"
                    href="/catalog-courses"
                  >
                    Рекомендации
                  </a>
                </Link>
              </Link>
            </HeaderNav>
          </nav>
          <AccountBar>
            <div
              className="accountBar"
            >
              <SearchInput>
                <div
                  className="searchInput"
                >
                  <form
                    className="searchForm"
                  >
                    <label
                      htmlFor="search"
                    >
                      <input
                        id="search"
                        placeholder="Search..."
                        type="text"
                      />
                    </label>
                    <button
                      onClick={[Function]}
                      type="submit"
                    />
                  </form>
                </div>
              </SearchInput>
              <ProfileButton>
                <div
                  className="wrapper"
                >
                  <div
                    className="dropdown"
                  >
                    <button
                      className="dropdownButton"
                      onClick={[Function]}
                      type="button"
                    />
                  </div>
                </div>
              </ProfileButton>
            </div>
          </AccountBar>
        </header>
      </div>
    </Header>
    <main
      className="container"
    >
      <PageHeader
        label="Изучаем JS"
      >
        <h2
          className="pageHeader"
        >
          Изучаем JS
        </h2>
      </PageHeader>
      <CourseInfo
        description="Узнайте, как создать высокопроизводительное веб-приложение с помощью react. Хуки, Redux, React Routing, useContext, Fireabase и др. Узнайте, как создать высокопроизводительное веб-приложение с помощью react. Хуки, Redux, React Routing, useContext, Fireabase и др. Узнайте, как создать высокопроизводительное веб-приложение с помощью react. Хуки, Redux, React Routing, useContext, Fireabase и др."
        info={
          Object {
            "authors": Array [
              "Ким Чен",
            ],
            "duration": "5ч 18м",
            "language": "Русский",
            "lastUpdate": "7/2020",
          }
        }
      >
        <div
          className="courseInfo"
        >
          <table
            className="courseTable"
          >
            <tbody>
              <tr
                className="courseRow"
              >
                <td
                  className="courseTerm"
                >
                  Авторы
                </td>
                <td
                  className="courseCell"
                >
                  Ким Чен
                </td>
              </tr>
              <tr
                className="courseRow"
              >
                <td
                  className="courseTerm"
                >
                  Язык
                </td>
                <td
                  className="courseCell"
                >
                  Русский
                </td>
              </tr>
              <tr
                className="courseRow"
              >
                <td
                  className="courseTerm"
                >
                  Обновлено
                </td>
                <td
                  className="courseCell"
                >
                  7/2020
                   
                </td>
              </tr>
              <tr
                className="courseRow"
              >
                <td
                  className="courseTerm"
                >
                  Продолжительность
                </td>
                <td
                  className="courseCell"
                >
                  5ч 18м
                </td>
              </tr>
            </tbody>
          </table>
          <p
            className="courseDescription"
          >
             
            Узнайте, как создать высокопроизводительное веб-приложение с помощью react. Хуки, Redux, React Routing, useContext, Fireabase и др. Узнайте, как создать высокопроизводительное веб-приложение с помощью react. Хуки, Redux, React Routing, useContext, Fireabase и др. Узнайте, как создать высокопроизводительное веб-приложение с помощью react. Хуки, Redux, React Routing, useContext, Fireabase и др.
             
          </p>
        </div>
      </CourseInfo>
      <section
        className="gallery"
      >
        <CourseCards
          cards={
            Array [
              Object {
                "id": 1,
                "imageSrc": "remote-assets/images/courses/bg-widget-m-1.jpg",
                "label": "React",
                "videoSrc": "",
              },
              Object {
                "id": 2,
                "imageSrc": "remote-assets/images/courses/bg-widget-m-2.jpg",
                "label": "CI/CD",
                "videoSrc": "",
              },
              Object {
                "id": 3,
                "imageSrc": "remote-assets/images/courses/bg-widget-m-3.jpg",
                "label": "Отрисовка на стороне сервера",
                "videoSrc": "",
              },
              Object {
                "id": 4,
                "imageSrc": "remote-assets/images/courses/bg-widget-m-4.jpg",
                "label": "ES2021",
                "videoSrc": "",
              },
              Object {
                "id": 5,
                "imageSrc": "remote-assets/images/courses/bg-widget-m-5.jpg",
                "label": "git",
                "videoSrc": "",
              },
            ]
          }
        >
          <ul
            className="catalogList"
          >
            <li
              className="cardContainer"
              key="React0"
            >
              <div
                className="catalogItem"
              >
                <SmartImage
                  path="remote-assets/images/courses/bg-widget-m-1.jpg"
                >
                  <img
                    className="cardImage"
                    onError={[Function]}
                    src="remote-assets/images/courses/bg-widget-m-1.jpg"
                  />
                </SmartImage>
                <a
                  className="catalogLink"
                  href="#"
                >
                  <h3
                    className="catalogHeader"
                  >
                    <span
                      className="catalogCategory"
                    >
                      React
                    </span>
                  </h3>
                </a>
              </div>
            </li>
            <li
              className="cardContainer"
              key="CI/CD1"
            >
              <div
                className="catalogItem"
              >
                <SmartImage
                  path="remote-assets/images/courses/bg-widget-m-2.jpg"
                >
                  <img
                    className="cardImage"
                    onError={[Function]}
                    src="remote-assets/images/courses/bg-widget-m-2.jpg"
                  />
                </SmartImage>
                <a
                  className="catalogLink"
                  href="#"
                >
                  <h3
                    className="catalogHeader"
                  >
                    <span
                      className="catalogCategory"
                    >
                      CI/CD
                    </span>
                  </h3>
                </a>
              </div>
            </li>
            <li
              className="cardContainer"
              key="Отрисовка на стороне сервера2"
            >
              <div
                className="catalogItem"
              >
                <SmartImage
                  path="remote-assets/images/courses/bg-widget-m-3.jpg"
                >
                  <img
                    className="cardImage"
                    onError={[Function]}
                    src="remote-assets/images/courses/bg-widget-m-3.jpg"
                  />
                </SmartImage>
                <a
                  className="catalogLink"
                  href="#"
                >
                  <h3
                    className="catalogHeader"
                  >
                    <span
                      className="catalogCategory"
                    >
                      Отрисовка на стороне сервера
                    </span>
                  </h3>
                </a>
              </div>
            </li>
            <li
              className="cardContainer"
              key="ES20213"
            >
              <div
                className="catalogItem"
              >
                <SmartImage
                  path="remote-assets/images/courses/bg-widget-m-4.jpg"
                >
                  <img
                    className="cardImage"
                    onError={[Function]}
                    src="remote-assets/images/courses/bg-widget-m-4.jpg"
                  />
                </SmartImage>
                <a
                  className="catalogLink"
                  href="#"
                >
                  <h3
                    className="catalogHeader"
                  >
                    <span
                      className="catalogCategory"
                    >
                      ES2021
                    </span>
                  </h3>
                </a>
              </div>
            </li>
            <li
              className="cardContainer"
              key="git4"
            >
              <div
                className="catalogItem"
              >
                <SmartImage
                  path="remote-assets/images/courses/bg-widget-m-5.jpg"
                >
                  <img
                    className="cardImage"
                    onError={[Function]}
                    src="remote-assets/images/courses/bg-widget-m-5.jpg"
                  />
                </SmartImage>
                <a
                  className="catalogLink"
                  href="#"
                >
                  <h3
                    className="catalogHeader"
                  >
                    <span
                      className="catalogCategory"
                    >
                      git
                    </span>
                  </h3>
                </a>
              </div>
            </li>
          </ul>
        </CourseCards>
      </section>
    </main>
  </DetailCourse>
</Provider>
`;
